examples/taskworkflow.go

```diff
@@ -0,0 +1,155 @@
+
+package main
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"syscall"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+	"github.com/firecracker-microvm/firecracker-containerd/proto"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+func main() {
+	var ip = flag.String("ip", "", "ip address assigned to the container")
+	log.SetFlags(log.Ldate | log.Ltime | log.Lmicroseconds)
+	flag.Parse()
+	if err := taskWorkflow(*ip); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func taskWorkflow(containerIP string) error {
+	log.Println("Creating contained client")
+	client, err := containerd.New("/run/containerd/containerd.sock")
+	if err != nil {
+		return errors.Wrapf(err, "creating client")
+
+	}
+	defer client.Close()
+	log.Println("Created contained client")
+
+	ctx := namespaces.WithNamespace(context.Background(), "firecracker-containerd-example")
+	image, err := client.Pull(ctx, "docker.io/library/nginx:latest",
+		containerd.WithPullUnpack,
+		containerd.WithPullSnapshotter("firecracker-naive"),
+	)
+	if err != nil {
+		return errors.Wrapf(err, "creating container")
+
+	}
+	log.Printf("Successfully pulled %s image\n", image.Name())
+	container, err := client.NewContainer(
+		ctx,
+		"demo",
+		containerd.WithSnapshotter("firecracker-naive"),
+		containerd.WithNewSnapshot("demo-snapshot", image),
+		containerd.WithNewSpec(
+			oci.WithImageConfig(image),
+			oci.WithHostNamespace(specs.NetworkNamespace),
+			oci.WithHostHostsFile,
+			oci.WithHostResolvconf,
+		),
+		containerd.WithRuntime("aws.firecracker", nil),
+	)
+	if err != nil {
+		return err
+	}
+	defer container.Delete(ctx, containerd.WithSnapshotCleanup)
+
+	task, err := container.NewTask(ctx,
+		cio.NewCreator(cio.WithStdio),
+		withFirecrackerConfig(containerIP),
+	)
+	if err != nil {
+		return errors.Wrapf(err, "creating task")
+
+	}
+	defer task.Delete(ctx)
+
+	log.Printf("Successfully created task: %s for the container\n", task.ID())
+	exitStatusC, err := task.Wait(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "waiting for task")
+
+	}
+
+	log.Println("Completed waiting for the container task")
+	if err := task.Start(ctx); err != nil {
+		return errors.Wrapf(err, "starting task")
+
+	}
+
+	log.Println("Successfully started the container task")
+	time.Sleep(3 * time.Second)
+
+	if containerIP != "" {
+		log.Println("Executing http GET on " + containerIP)
+		getResponse(containerIP)
+	}
+
+	if err := task.Kill(ctx, syscall.SIGTERM); err != nil {
+		return errors.Wrapf(err, "killing task")
+
+	}
+	status := <-exitStatusC
+	code, _, err := status.Result()
+	if err != nil {
+		return errors.Wrapf(err, "getting task's exit code")
+
+	}
+	log.Printf("task exited with status: %d\n", code)
+	return nil
+}
+
+func withFirecrackerConfig(containerIP string) containerd.NewTaskOpts {
```

Comment:
Should this be an exported functional option somewhere?  Maybe one that takes the MAC and HostDevName as arguments as well?

Reply:
May be. But, I don't feel the do it right now since we'd need to hardcode the n/w interface name anyway, unless we accept it as a flag for the program. It's probably fine the way it is for this example. wdyt?

Reply:
Sounds good.